src/crypto/verify.c

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: ISC
+// SPDX-FileCopyrightText: 2013-2024 Frank Denis <j at pureftpd dot org>
+
+#include <stdint.h>
+
+#include "verify.h"
+
+#define crypto_verify_32_BYTES 32U
+
+#if defined(__x86_64__) && defined(__SSE2__)
+
+# ifdef __GNUC__
+#  pragma GCC target("sse2")
+# endif
+# include <emmintrin.h>
+
+static inline int
+crypto_verify_n(const unsigned char *x_, const unsigned char *y_,
+                const int n)
+{
+    const    __m128i zero = _mm_setzero_si128();
```

Comment:
Worth noting: This very much requires strict aliasing disabled (which sodium does, quite explicitly, and by design; whereas this project's CMakeLists.txt suggest an intent to go the other direction):
```
  # With GCC 6.1.1 the compiled binary malfunctions due to aliasing. Until that
  # is fixed in the code (Issue #847), force compiler to be conservative.
  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-strict-aliasing")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-aliasing")
```

The more creative the compiler vendors get, the more problematic even a single vendored function like this becomes.